packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -413,17 +414,25 @@ export abstract class AbstractBackgroundJobProcessor<
     const jobOptsRemoveOnComplete = job.opts.removeOnComplete
     if (jobOptsRemoveOnComplete === true || jobOptsRemoveOnComplete === 1) return
 
-    // @ts-ignore
-    const updateDataPromise = job.updateData({ metadata: job.data.metadata })
-    const clearLogsPromise = job.clearLogs()
+    const updateDataPromise = job
+      // @ts-ignore
+      .updateData({ metadata: job.data.metadata })
+      .finally(() => this.runningPromises.delete(updateDataPromise))
 
-    const updateDataPromiseIndex = this.runningPromises.push(updateDataPromise) - 1
-    const clearLogsPromiseIndex = this.runningPromises.push(clearLogsPromise) - 1
+    this.runningPromises.add(updateDataPromise)
 
-    await Promise.all([updateDataPromise, clearLogsPromise]).then(() => {
-      this.runningPromises.splice(updateDataPromiseIndex, 1)
-      this.runningPromises.splice(clearLogsPromiseIndex, 1)
-    })
+    const clearLogsPromise = job
+      .clearLogs()
+      .finally(() => this.runningPromises.delete(clearLogsPromise))
+
+    this.runningPromises.add(clearLogsPromise)
+
+    try {
+      await Promise.all([updateDataPromise, clearLogsPromise])
+    } catch (error) {
```

Comment:
you may want to use `allSettled` instead, if you want to wait for all promises to resolve regardless of how successful that is